app/controllers/concerns/batch_import_controller_behavior.rb

```diff
@@ -24,23 +34,41 @@ def self.form_class(_ = nil)
 
   protected
 
+    def build_form
+      super
+      @form.payload_concern = params[:payload_concern] || Hyrax.primary_work_type.model_name.name
+    end
+
+    def handle_payload_concern!
+      unsafe_pc = params.fetch(:batch_import_item, {})[:payload_concern]
+      # Calling constantize on user params is disfavored (per brakeman),
+      # so we sanitize by matching it against an authorized model.
+      safe_pc = Hyrax::SelectTypeListPresenter
+                .new(current_user).authorized_models.map(&:to_s).find { |x| x == unsafe_pc }
+      raise CanCan::AccessDenied, "Cannot create an object of class '#{unsafe_pc}'" unless safe_pc
+      # authorize! :create, safe_pc
```

Comment:
Just curious as to why this commented out line is here. Was this from upstream hyrax?